app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -714,8 +714,10 @@ class BrowserFragment : Fragment(), BackHandler, CoroutineScope {
             viewModel?.tabs = listOf(tabs)
             val selectedSet = mutableSetOf(tabs)
             viewModel?.selectedTabs = selectedSet
-            viewModel?.saveCollectionStep = SaveCollectionStep.SelectCollection
             viewModel?.tabCollections = requireComponents.core.tabCollectionStorage.cachedTabCollections.reversed()
+            viewModel?.saveCollectionStep =
+                if (viewModel?.tabCollections?.size == 0) SaveCollectionStep.NameCollection else
```

Comment:
This logic is repeated three times and it's just slightly different in the other two places.
```if (viewModel.tabCollections.isEmpty())```
Could we create a method and reuse it?

Reply:
I added an extension method on the List of tab collections